src/flashbar/common.tsx

```diff
@@ -140,3 +144,48 @@ export function useFlashbar({
     handleFlashDismissed,
   };
 }
+
+// Hook for managing flashbar items visibility with persistence
+export function useFlashbarVisibility(items: ReadonlyArray<FlashbarProps.MessageDefinition>) {
+  const [checkedPersistenceKeys, setCheckedPersistenceKeys] = useState<Set<string>>(new Set());
+  const [persistentItemsVisibility, setPersistentItemsVisibility] = useState<Map<string, boolean>>(new Map());
+
+  const visibleItems = useMemo(() => {
+    return items.filter(
+      item =>
+        !item.persistenceConfig?.uniqueKey || persistentItemsVisibility.get(item.persistenceConfig.uniqueKey) === true
+    );
+  }, [items, persistentItemsVisibility]);
+
+  useEffect(() => {
+    const newPersistentItems = items.filter(
+      item => item.persistenceConfig?.uniqueKey && !checkedPersistenceKeys.has(item.persistenceConfig.uniqueKey)
+    );
+
+    if (newPersistentItems.length === 0) {
+      return;
+    }
+
+    const checkNewPersistentItems = async () => {
+      const newVisibility = new Map(persistentItemsVisibility);
+      const newKeys = new Set(checkedPersistenceKeys);
+
+      await Promise.all(
+        newPersistentItems.map(async item => {
+          const isDismissed = await retrieveFlashbarDismiss(item.persistenceConfig!);
+          newVisibility.set(item.persistenceConfig!.uniqueKey, !isDismissed);
+          newKeys.add(item.persistenceConfig!.uniqueKey);
+        })
+      );
+
+      setPersistentItemsVisibility(newVisibility);
```

Comment:
⛳  What happens of items list changes multiple times and there are several async `checkNewPersistentItems` calls happening in parallel? They will overwrite each other results.

It would be good to have a unit test on this as well